contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+  string public data;
+
+  bool public hasPaidOut;
+
+  Fulfillment[] public fulfillments;
+
+  Contribution[] public contributions;
+
+
+
+  /*
+   * Events
+   */
+  event BountyFulfilled(uint256 _fulfillmentId);
+  event FulfillmentUpdated(uint _fulfillmentId);
+  event FulfillmentAccepted(uint256 _fulfillmentId);
+  event BountyDrained(address controller);
+  event BountyChanged();
+
+  /*
+   * Structs
+   */
+  struct Fulfillment {
+      address[] fulfillers;
+      uint[] numerators;
+      uint denomenator;
+      string data;
+      bool accepted;
+  }
+
+  struct Contribution {
+      address contributor;
+      uint[] amounts;
+      StandardToken[] tokens;
+      bool refunded;
+      mapping (address => bool) hasContributed;
+  }
+
+  /*
+   * Modifiers
+   */
+
+  modifier onlyController() {
+      require(msg.sender == controller);
+      _;
+  }
+
+  modifier validateFulfillmentArrayIndex(uint _index) {
+      require(_index < fulfillments.length);
+      _;
+  }
+
+  modifier validateNotTooManyFulfillments(){
+    require((fulfillments.length + 1) > fulfillments.length);
+    _;
+  }
+
+  modifier notController(address _fulfiller) {
+      require(_fulfiller != controller);
+      _;
+  }
+
+  modifier hasNotPaid(){
+      require(!hasPaidOut);
+      _;
+  }
+
+  modifier onlyContributor(uint _contributionId){
+      require(contributions[_contributionId].contributor == msg.sender);
+      _;
+  }
+
+  modifier notYetRefunded(uint _contributionId){
+      require(!contributions[_contributionId].refunded);
+      _;
+  }
+
+  modifier fulfillmentNotYetAccepted(uint _fulfillmentId){
+      require(!fulfillments[_fulfillmentId].accepted);
+      _;
+  }
+
+  modifier sameLength(uint l1, uint l2){
+      require(l1 == l2);
+      _;
+  }
+
+  modifier sumToOne(uint[] _numerators, uint _denomenator){
+      uint sum = 0;
+      for (uint i = 0; i < _numerators.length; i++){
+        sum += _numerators[i];
+      }
+      require(sum == _denomenator);
+      _;
+  }
+
+
+  /*
+   * Public functions
+   */
+
+  function ()
+  public
+  payable
+  {
+
+  }
+
+  function initializeBounty(
+      address _controller,
+      string _data)
+      public
+  {
+    require(controller == address(0));
+    require(_controller != address(0));
+    // an controller of a bounty is only 0x0 when it is uninitialized,
+    // so this check prevents initialization from being called multiple times
+
+    controller = _controller;
+    data = _data;
+
+  }
+
+  function refundableContribute(uint[] _amounts, StandardToken[] _tokens)
+  public
+  payable
+  {
+      require(_amounts.length == _tokens.length);
+
+      contributions.push(Contribution(msg.sender, _amounts, _tokens, false));
+      uint contributionId = contributions.length - 1;
+      for (uint i = 0; i < _amounts.length; i++){
+          require(!contributions[contributionId].hasContributed[_tokens[i]]);
+          contributions[contributionId].hasContributed[_tokens[i]] = true;
```

Comment:
do we care if they just put zero in this value?

Reply:
issue

Reply:
also remember to add issue for discussion around event listening for multiplec ontracts)